templates/aws-tap-entrypoint-existing-vpc.template.yaml

```diff
@@ -1374,11 +1361,21 @@ Resources:
                   - eks:DescribeCluster
                   - eks:DescribeNodegroup
                 Resource:
-                  - !Sub arn:${AWS::Partition}:eks:${AWS::Region}:${AWS::AccountId}:cluster/${EKSClusterName}
-                  - !Sub arn:${AWS::Partition}:eks:${AWS::Region}:${AWS::AccountId}:cluster/${EKSClusterName}-run
-                  - !Sub arn:${AWS::Partition}:eks:${AWS::Region}:${AWS::AccountId}:cluster/${EKSClusterName}-build
-                  - !Sub arn:${AWS::Partition}:eks:${AWS::Region}:${AWS::AccountId}:cluster/${EKSClusterName}-iterate
-                  - !Sub arn:${AWS::Partition}:eks:${AWS::Region}:${AWS::AccountId}:cluster/${EKSClusterName}-view
+                  - !Sub
+                      - arn:${AWS::Partition}:eks:${AWS::Region}:${AWS::AccountId}:cluster/${clusterName}
+                      - clusterName: !Join ['-', ['tap', !Select [4, !Split ['-', !Select [2, !Split ['/', !Ref AWS::StackId]]]]]]
+                  - !Sub
+                      - arn:${AWS::Partition}:eks:${AWS::Region}:${AWS::AccountId}:cluster/${clusterName}
+                      - clusterName: !Join ['-', ['tap', !Select [4, !Split ['-', !Select [2, !Split ['/', !Ref AWS::StackId]]]], 'run']]
```

Comment:
Repeating this same Join/Split multiple times might cause maintenance bugs.
Suggest using Mappings to define variable names. something like
```

"Mappings" : {
  "CustomVariables" : {
    "Variable1" : { "Value" : "foo" },
    "Variable2" : { "Value" : "bar" }
  }
}
```

Reply:
That has been discussed in #93 and (in parts) in my latest slack message:
- `!FindImMap` does only support a very limited set of intrinsic functions
- `Mappings` don't allow for intrinsic funtions

One way I can see where we could solve some of those issues might be the [AWS::LanguageExtensions transform] and the enhanced version of `!FindInMap`, however:
- I am hesitant to introduce that this late in the cycle without a deeper understanding of the [AWS::LanguageExtensions transform] and potential side-effects
- [AWS::LanguageExtensions transform]/the enhance version of `!FindInMap` might not even work because [this](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-findinmap-enhancements.html#:~:text=as%20long%20as%20it%27s%20able%20to%20resolve%20to%20a%20valid%20value%20during%20the%20transform)

So this is the (IMHO) the easiest thing we came up with.

I do agree tho, that we should look into refactoring all of that, but that then is a broader activity, because that will also bleed into the cloud-init/setup-script.

@aaronmeza @satya-dillikar  


[AWS::LanguageExtensions transform]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/transform-aws-languageextensions.html

Reply:
To rephrase that: There is no way I am aware of to get "the random bits" into the mapping or make it easier to add them while pulling values out of the mapping via `!FindInMap`.

Reply:
> [...] not worth exploring more [...]
> [...] fixed-hard-coded names [...] are ok for us as all clusters are in a new vpc [...]

I am not sure I understand your comment. Are you suggesting to maybe *not* merge this PR, or any PR making the cluster name somewhat unique?

FWIW: "The name must be unique within the AWS Region and AWS account that you're creating the cluster in" ([here][uniq])


[uniq]: https://docs.aws.amazon.com/eks/latest/userguide/create-cluster.html#:~:text=The%20name%20must%20be%20unique%20within%20the%20AWS%20Region%20and%20AWS%20account%20that%20you%27re%20creating%20the%20cluster%20in